linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -361,22 +361,25 @@ object Infos {
       .addInterfaces(classDef.interfaces.map(_.name))
 
     classDef.memberDefs foreach {
-      case fieldDef: FieldDef =>
+      case fieldDef: AnyFieldDef =>
         builder.maybeAddReferencedFieldClass(fieldDef.ftpe)
 
       case methodDef: MethodDef =>
         builder.addMethod(generateMethodInfo(methodDef))
 
-      case propertyDef: PropertyDef =>
-        builder.addMethod(generatePropertyInfo(propertyDef))
+      case methodDef: JSMethodDef =>
+        builder.addExportedMember(generateJSMethodInfo(methodDef))
+
+      case propertyDef: JSPropertyDef =>
+        builder.addExportedMember(generateJSPropertyInfo(propertyDef))
     }
 
     if (classDef.topLevelExportDefs.nonEmpty) {
       builder.setIsExported(true)
 
       val optInfo = generateTopLevelExportsInfo(classDef.name.name,
           classDef.topLevelExportDefs)
-      optInfo.foreach(builder.addMethod(_))
+      optInfo.foreach(builder.addTopLevelExportedMember(_))
```

Comment:
Yes, it's intentional. At some point I'll also change `generateTopLevelExportsInfo` so that it exports one (or zero) reachability info *per export*, rather than one for all. It doesn't really make sense to use one for all; it's only for historical reasons of how infos were built.

However changing that is not required for this commit, so I left it for later (also it's a completely internal change).